src/marshmallow_annotations/converter.py

```diff
@@ -13,16 +13,16 @@
 from .registry import registry
 
 NoneType = type(None)
-_UNION_TYPE = type(Union)
 
 
 def _is_optional(typehint):
     # only supports single type optionals/unions
     # as for the implementation... look, don't ask me
     return (
-        _UNION_TYPE in type(typehint).__mro__  # noqa
-        and len(typehint.__args__) == 2  # noqa
-        and typehint.__args__[1] is NoneType  # noqa
+        hasattr(typehint, '__origin__') and
+        typehint.__origin__ is Union and
+        len(typehint.__args__) == 2 and
+        NoneType in typehint.__args__
```

Comment:
Hm. I wish the docs were better about interacting with typehints and/or there were facilities builtin to help with this sort of code generation. With dataclasses in 3.7 I'm sure the core devs felt this pain too.